app/kumactl/pkg/cmd/root_context.go

```diff
@@ -105,6 +111,21 @@ func (rc *RootContext) CurrentDataplaneOverviewClient() (kumactl_resources.Datap
 	return rc.Runtime.NewDataplaneOverviewClient(controlPlane.Coordinates.ApiServer)
 }
 
+func (rc *RootContext) CurrentDpTokenClient() (tokens.DpTokenClient, error) {
+	controlPlane, err := rc.CurrentControlPlane()
+	if err != nil {
+		return nil, err
+	}
+	// this will be replaced by inferring addresses https://github.com/Kong/kuma/issues/315
+	apiServerUrl, err := url.Parse(controlPlane.Coordinates.ApiServer.Url)
+	if err != nil {
+		return nil, err
+	}
+	port := apiServerUrl.Port()
+	const defaultDpTokenPort = 5679
+	return rc.Runtime.NewDpTokenClient(strings.ReplaceAll(controlPlane.Coordinates.ApiServer.Url, port, strconv.Itoa(defaultDpTokenPort)))
```

Comment:
alternative to `strings.ReplaceAll(..)` would be `net.SplitHostPort(..)`

Reply:
I was looking into it but it doesn't play well when dealing with URL like `http://127.0.0.1:1234`
https://play.golang.org/p/azgRBVrNZiA

Reply:
You shouldn't pass the entire URL to `net.SplitHostPort(..)`. Just `url.Host`

E.g., https://play.golang.org/p/fvEQYu_-99b